import/popups/DownloadMenuItem.qml

```diff
@@ -48,4 +49,28 @@ MenuItem {
                                       })
         }
     }
+
+    onClicked: {
+        if (WebEngineSettings.useDownloadDir) {
+            root.targetDirectory = WebEngineSettings.downloadDir
+            startDownload()
+        } else {
+            pageStack.animatorPush(folderPickerPage)
+        }
+    }
+
+    Component {
+        id: folderPickerPage
+
+        FolderPickerPage {
```

Comment:
Can the folder picker be set to start in the most appropriate place (`StandardPaths.download`, or the last selected folder as appropriate)?

Reply:
There is a problem, when you initialize rootPath for FolderPickerDIalog, you can't go to back. It limits a user  to save any directory above last selected folder

Reply:
Yes, I recall this being a limitation of FolderPicker. I think this is something that will need to be addressed, as it's part of the design: 'By default "Downloads" directory should be shown, but when user chooses another directory and saves the file to it, **next time browser should offer this directory first**' (my emphasis).

What would you propose? We can fix FolderPicker first then return to this, or merge this, fix FolderPicker, then create a new PR to make changes here?